### PR TITLE
Support for german locale on negative amounts

### PR DESCRIPTION
**Pull request by KevinHansen23**
Currently if locale is set to de_DE negative Amounts are formatted with `,` instead of `.`. That leads to a NumberFormatException, which is fixed with these changes.

### CODE REVIEW

**Comment by KevinHansen23** [comment]
Hi @eiriarte-mendez,

any Updates on this?

Kind regards
Kevin Hansen

**Comment by eiriarte-mendez** [comment]
Hi @KevinHansen23, 

Thanks for the PR. Could you please add an explanation why this library should support other number formats? Our Gateway at RatePAY accepts only the english format at the moment...

**Review by eiriarte-mendez — COMMENTED** [review]
Could you please update your PR to the latest master, so test can be executed

**Comment by KevinHansen23** [comment]
Hi @eiriarte-mendez,

our CRM is set to `(LC_ALL,"de_DE.utf8");`.

The problem now is, that the library only accepts the english number format but uses `floatval();` which formats numbers based on the current locale. So even if we push a value with the correct format (e.g. `5.95`), the library calls `changeValueToNegative` and the value is changed to `-5,95`.

There is no way to fix this without these changes because the malformatting happens within the library.
In the `changeAmountToFloat` function you addressed this problem, but not within the `changeValueToNegative` function.

Regarding the Update:
I didn't find a way to update my fork with an open PR (I'm using Github in browser only).

Kind regards
Kevin

**Review by eiriarte-mendez — CHANGES_REQUESTED** [review]
PR is not passing checks

**Comment by MartinKuhl** [comment]
> PR is not passing checks

what has to be adjust?

File: src/Service/Util.php [hunk 1]
```diff
@@ -18,24 +18,25 @@
 class Util
 {
     /**
-     * Changes value to negative if necessary.
+     * Changes value to negative if necessary
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:21]:
```suggestion
     * Changes value to negative if necessary.
```

File: src/Service/Util.php [hunk 2]
```diff
@@ -18,24 +18,25 @@
...
+
+        $value = number_format($value, 2, ".", "");
 
-        return ($value > 0) ? $value * -1 : $value;
+        return $value;
     }
 
     /**
-     * Changes the camel case notation to separation by underscore.
+     * Changes the camel case notation to separation by underscore
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:37]:
```suggestion
     * Changes the camel case notation to separation by underscore.
```

File: src/Service/Util.php [hunk 3]
```diff
@@ -18,24 +18,25 @@
...
      * @param $string
-     *
      * @return string
      */
     public static function changeValueToNegative($value)
     {
         $value = floatval($value);
+        $value =  ($value > 0) ? $value * -1 : $value;
+
+        $value = number_format($value, 2, ".", "");
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:31]:
```suggestion
        $value = number_format($value, 2, '.', '');
```

File: src/Service/Util.php [hunk 4]
```diff
@@ -18,24 +18,25 @@
...
+        return $value;
     }
 
     /**
-     * Changes the camel case notation to separation by underscore.
+     * Changes the camel case notation to separation by underscore
      *
      * @param $string
-     *
      * @return string
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:40]:
```suggestion
     *
     * @return string
```

File: src/Service/Util.php [hunk 5]
```diff
@@ -18,24 +18,25 @@
 class Util
 {
     /**
-     * Changes value to negative if necessary.
+     * Changes value to negative if necessary
      *
      * @param $string
-     *
      * @return string
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:24]:
```suggestion
     *
     * @return string
```

File: src/Service/Util.php [hunk 6]
```diff
@@ -18,24 +18,25 @@
...
      */
     public static function changeValueToNegative($value)
     {
         $value = floatval($value);
+        $value =  ($value > 0) ? $value * -1 : $value;
```
Comment (eiriarte-mendez) [inline src/Service/Util.php:29]:
```suggestion
        $value = ($value > 0) ? $value * -1 : $value;
```

**Review by eiriarte-mendez — COMMENTED** [review]
only style checks...

**Pull request closed without merging**